wallet-new/integration/AccountSpecs.hs

```diff
@@ -98,6 +100,68 @@ accountSpecs wRef wc =
 
             map (AccountBalance . accAmount) accsUpdated `shouldBe` balancesPartialUpdated
 
+
+        prop "redeeming avvm key gives rise to the corresponding increase of balance of wallet'account - mnemonic not used" $ withMaxSuccess 1 $
+            monadicIO $ do
+
+            newWallet <- run $ randomWallet CreateWallet
+            Wallet{..} <- run $ createWalletCheck wc newWallet
+
+            --adding new account
+            rAcc <- pick arbitrary :: PropertyM IO NewAccount
+            newAcctResp <- run $ postAccount wc walId rAcc
+            newAcct <- run $ wrData <$> newAcctResp `shouldPrism` _Right
+
+            balancePartialRespB <- run $ getAccountBalance wc walId (accIndex newAcct)
+            balancesPartialB <- run $ wrData <$> balancePartialRespB `shouldPrism` _Right
+            let zeroBalance = AccountBalance $ V1 (Core.mkCoin 0)
+            liftIO $ balancesPartialB `shouldBe` zeroBalance
+
+            -- state-demo/genesis-keys/keys-fakeavvm/fake-9.seed
+            let avvmKey = "QBYOctbb6fJT/dBDLwg4je+SAvEzEhRxA7wpLdEFhnY="
+
+            --password is set to Nothing in the current implementation of randomWallet
+            --when it changes redemptionSpendingPassword handles it, otherwise passPhare addresses it
+            passPhrase <- pure mempty :: PropertyM IO SpendingPassword
+            let redemption = Redemption
+                    { redemptionRedemptionCode = ShieldedRedemptionCode avvmKey
+                    , redemptionMnemonic = Nothing
+                    , redemptionSpendingPassword = case newwalSpendingPassword newWallet of
+                            Just spPassw -> spPassw
+                            Nothing      -> passPhrase
+                    , redemptionWalletId = walId
+                    , redemptionAccountIndex = accIndex newAcct
+                    }
+
+            etxn <- run $ redeemAda wc redemption
+
+            txn <- run $ fmap wrData etxn `shouldPrism` _Right
+
+            liftIO $ threadDelay 90000000
+
+            --checking if redemption give rise to transaction indexing
+            eresp <- run $ getTransactionIndex
+                wc
+                (Just walId)
+                (Just (accIndex newAcct))
+                Nothing
+            resp <- run $ fmap wrData eresp `shouldPrism` _Right
+            liftIO $ map txId resp `shouldContain` [txId txn]
+
+            --balance for the previously zero-balance account should increase by 100000
+            balancePartialResp <- run $ getAccountBalance wc walId (accIndex newAcct)
+            balancesPartial <- run $ wrData <$> balancePartialResp `shouldPrism` _Right
+            let nonzeroBalance = AccountBalance $ V1 (Core.mkCoin 100000)
+            liftIO $ balancesPartial `shouldBe` nonzeroBalance
+
+            --redeemAda for the same redeem address should result in error
+            etxnAgain <- run $ redeemAda wc redemption
```

Comment:
Would have been nice to have a separated test for that as we're testing a different requirement with this :+1: